extensions/amp-story/1.0/amp-story.js

```diff
@@ -1624,8 +1633,35 @@ export class AmpStory extends AMP.BaseElement {
    * @visibleForTesting
    */
   onResize() {
-    const uiState = this.getUIType_();
-    this.storeService_.dispatch(Action.TOGGLE_UI, uiState);
+    const previousUiState = this.uiState_;
+    this.uiState_ = this.getUIType_();
+
+    const wasMobile = previousUiState === UIType.MOBILE;
+    const inputHasFocus = this.win.document.activeElement?.tagName === 'INPUT';
+    const softKeyboardIsProbablyOpen = wasMobile && inputHasFocus;
+    if (!softKeyboardIsProbablyOpen) {
+      // The opening of the Android soft keyboard triggers a viewport resize
+      // that can cause the story's dimensions to appear to be those of a
+      // desktop. Here, we assume that the soft keyboard is open if the story
+      // has resized while an input element has focus, and we do not toggle the
+      // UI.
```

Comment:
Looks like this logic + comment belong in `getUIType_()` and not here, that should return `MOBILE` when the soft keyboard is open. Any other caller would get a wrong `UIType` with this implementation